### DISCUSSION TITLE
Why I cannot send/set parameters to PopupWindow

### DISCUSSION THREAD

**Discussion by qarmin** [slint-ui/slint#3886] [Q&A]
In https://slint.dev/releases/1.2.2/docs/slint/src/language/builtins/elements#popupwindow I see this line
```
Note: It isn’t allowed to access properties of elements within the popup from outside of the PopupWindow.
```

which is really strange.
I can use from PopupWindow external callbacks, but why exactly I cannot access PopupWindow properties? 
How can I send data to this item?
Storing data outside PopupWindow, that only PopupWindow uses is not really fun to maintain.

With qml I use such construction to send data to popup
```
var popupComponent = Qt.createComponent("qrc:/TestPopup.qml")
var popup2 = popupComponent.createObject(window, {"internal_property" : data});
popup2.open()
```

**Comment by ogoffart** [comment]
This is currently a limitation which is caused by the current implementation of the PopupWindow.
The PopupWindow is only being allocated when it is shown. Therefore one can't access inner properties from the outside of it.

There are a bunch of issue with the PopupWindow which are tracked in the meta issue https://github.com/slint-ui/slint/issues/1143
